src/client-scopes/details/SearchFilter.tsx

```diff
@@ -0,0 +1,188 @@
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import {
+  Dropdown,
+  DropdownItem,
+  DropdownToggle,
+  Select,
+  SelectOption,
+  ToolbarItem,
+} from "@patternfly/react-core";
+import { FilterIcon } from "@patternfly/react-icons";
+
+import {
+  AllClientScopes,
+  clientScopeTypesSelectOptions,
+} from "../../components/client-scope/ClientScopeTypes";
+import type { Row } from "../../clients/scopes/ClientScopes";
+
+export type SearchType = "name" | "type" | "protocol";
+export enum ProtocolType {
+  all = "all",
+  saml = "saml",
+  openid = "openid-connect",
+}
+type ProtocolTypeKeys = keyof typeof ProtocolType;
+
+export const nameFilter =
+  (search: string = "") =>
```

Comment:
No need to specify the type if a default is provided from which the type can be inferred.
```suggestion
  (search = "") =>
```